### PR TITLE
Add upgrade pairs to upgrade type

### PR DESCRIPTION
**Pull request by spiderbiggen**
Change `upgrade_type(<ID>, <SourceFunctionID>, <TargetFunctionID>)` to `upgrade_type(<ID>, [upgrade_pair(<SourceFunctionID>, <TargetFunctionID>)...])` to support multiple pairs per upgrade

### CODE REVIEW

File: contextvh/src/test/java/contextvh/translators/J2UpgradeTypeTest.java [hunk 1]
```diff
@@ -24,23 +26,24 @@
...
 	 */
-	private J2UpgradeType translator;
+	private Translator translator = Translator.getInstance();
 	/**
 	 * The indicators to translate.
 	 */
 	private UpgradeType upgradeType;
 	private UpgradePair upgradePair;
-	private ArrayList<UpgradePair>  a;
+	private List<UpgradePair> a;
```
Comment (klikooo) [inline contextvh/src/test/java/contextvh/translators/J2UpgradeTypeTest.java]: Maybe change to a more descriptive name?

**Comment by spiderbiggen** [comment]
travis buld log is [here](https://travis-ci.org/spiderbiggen/tygron/builds/135965489)

File: contextvh/src/test/java/contextvh/translators/J2UpgradeTypeTest.java (hunk 1 above)
Comment (spiderbiggen) [inline contextvh/src/test/java/contextvh/translators/J2UpgradeTypeTest.java]: fair point

**Comment by spiderbiggen** [comment]
new travis build log is here https://travis-ci.org/spiderbiggen/tygron/builds/135969619

**Comment by dionytadema** [comment]
conflicts :(

**Comment by MHMCD** [comment]
[Build Status](https://travis-ci.org/spiderbiggen/tygron)

Easier to check travis build logs

**Comment by spiderbiggen** [comment]
fixed merge conflicts

**Comment by jannelie** [comment]
we approve -CrashTestDummies

**Comment by hmyeh** [comment]
we have looked statically at the code and approve of that - Danshal

**Comment by MHMCD** [comment]
We approve

\- CodeFox [CodeFox](https://github.com/codefoxnl)

**Comment by MarcianoJ** [comment]
Code looks OK, was this also tested with the agent?

**Comment by spiderbiggen** [comment]
Marciano yes it was

File: contextvh/src/main/java/contextvh/translators/J2UpgradeType.java [hunk 2]
```diff
@@ -21,18 +20,17 @@
 
     /**
      * Translates the UpgradeType object in the form of:
-     * upgrade_type(<ID>, <SourceFunctionID>, <TargetFunctionID>) or
-     * upgrade_type(<ID>) if there is no pair available.
+     * {@code upgrade_type(<ID>, <SourceFunctionID>, <TargetFunctionID>)} or
+     * {@code upgrade_type(<ID>)} if there is no pair available.
      */
     @Override
     public Parameter[] translate(final UpgradeType u) throws TranslationException {
```
Comment (klikooo) [inline contextvh/src/main/java/contextvh/translators/J2UpgradeType.java]: Does checkstyle pass with such a short variable name?

**Comment by tvleest** [comment]
This can probably be merged now?

**Comment by MaxGroenenboom** [comment]
I think the note @klikooo placed should be addressed.

**Comment by tvleest** [comment]
Ah overlooked that one.
According to travis it passed, but it shouldn't? 
The name isn't changed by this PR though, but we could change it.

**Comment by spiderbiggen** [comment]
Checkstyle passes, but as @tvleest said we didn't change this it was already that way when i started working on it

File: contextvh/src/main/java/contextvh/translators/J2UpgradePair.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    private final Translator translator = Translator.getInstance();
+
+    /**
+     * Translates the UpgradeType object in the form of:
+     * {@code upgrade_type(<ID>, <SourceFunctionID>, <TargetFunctionID>)} or
+     * {@code upgrade_type(<ID>)} if there is no pair available.
+     */
+    @Override
+    public Parameter[] translate(final UpgradePair u) throws TranslationException {
```
Comment (klikooo) [inline contextvh/src/main/java/contextvh/translators/J2UpgradePair.java]: Change variable name to a more descriptive one

**Comment by tvleest** [comment]
Changed variable names. If there aren't any other concerns, we can merge.

**Comment by MaxGroenenboom** [comment]
Looks good, I approve ;)

**Pull request merged**